### PR TITLE
Fix person event role association

### PR DESCRIPTION
**Pull request by kaoxi998533**

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/EditCommand.java [hunk 1]
```diff
@@ -97,10 +98,10 @@ private static Person createEditedPerson(Person personToEdit, EditPersonDescript
         Name updatedName = editPersonDescriptor.getName().orElse(personToEdit.getName());
         Phone updatedPhone = editPersonDescriptor.getPhone().orElse(personToEdit.getPhone());
         Email updatedEmail = editPersonDescriptor.getEmail().orElse(personToEdit.getEmail());
-        Set<Event> updatedEvents = editPersonDescriptor.getEvents().orElse(personToEdit.getEvents());
-        Set<Role> updatedRoles = editPersonDescriptor.getRoles().orElse(personToEdit.getRoles());
+        Map<Event, Set<Role>> updatedEventRoles =
```
Comment (leowolf275) [inline src/main/java/seedu/address/logic/commands/EditCommand.java:101]: Great idea to use a map to connect the events with the roles!

File: src/main/java/seedu/address/logic/parser/ParserUtil.java [hunk 2]
```diff
@@ -273,27 +305,17 @@ public static Role parseRole(String role) throws ParseException {
...
+     * Leading and trailing whitespaces will be trimmed.
+     *
+     * @throws ParseException if any of the given events are invalid.
      */
-
-    public static Set<Event> parseEvents(Collection<String> events) throws ParseException {
+    public static List<Event> parseEvents(List<String> events) throws ParseException {
         requireNonNull(events);
-        final Set<Event> eventSet = new HashSet<>();
+        final List<Event> eventList = new ArrayList<>();
```
Comment (leowolf275) [inline src/main/java/seedu/address/logic/parser/ParserUtil.java:315]: Ensure that this line is covered by tests later!

File: src/test/data/JsonSerializableAddressBookTest/duplicatePersonAddressBook.json [hunk 3]
```diff
@@ -3,25 +3,15 @@
     "name": "Alice Pauline",
     "phone": "94351253",
     "email": "alice@example.com",
-    "personEvents": [{
-      "name": "IFG"
-    }],
-    "roles": [ "friends" ]
+    "eventRoles": {
+      "IFG": ["friends"]
```
Comment (leowolf275) [inline src/test/data/JsonSerializableAddressBookTest/duplicatePersonAddressBook.json:7]: It is nice to see that you updated the JSON format for the Addressbook to account for the new structure!

**Review by leowolf275 — APPROVED** [review]
LGTM!

File: src/test/data/JsonSerializableAddressBookTest/duplicatePersonAddressBook.json [hunk 4]
```diff
@@ -3,25 +3,15 @@
...
-    }]
-  } ],
-  "events": [ {
-    "name": "IFG"
-  }, {
-    "name": "AUG"
-  }, {
-    "name": "SUNIG"
-  }, {
-    "name": "IVP"
```
Comment (youngseopark05) [inline src/test/data/JsonSerializableAddressBookTest/duplicatePersonAddressBook.json:25]: We still need the "events" row to show the events separately on the right panel as we discussed, can we not delete this one?

**Review by youngseopark05 — CHANGES_REQUESTED** [review]
The tests seem really comprehensive, and I just made some clarifying remarks so the team can see what changes were made. 

Just for the JSON files, we do need to keep the outer "events" so we can display them separately on the right panel.

File: src/main/java/seedu/address/storage/JsonSerializableAddressBook.java [hunk 5]
```diff
@@ -22,11 +22,15 @@ class JsonSerializableAddressBook {
     private final List<JsonAdaptedPerson> persons = new ArrayList<>();
 
     @JsonCreator
-    public JsonSerializableAddressBook(@JsonProperty("events") List<JsonAdaptedEvent> events,
```
Comment (youngseopark05) [inline src/main/java/seedu/address/storage/JsonSerializableAddressBook.java:25]: We also need to keep this to list the events separately.

**Review by youngseopark05 — CHANGES_REQUESTED** [review]
Adding on to my previous point about the JSON files still needing the events for the rightside panel, I think we should make sure JsonSerializableAddressBook keeps the separate events List as well.

**Pull request closed without merging**